### PR TITLE
[Enhancement/YataiService] Introduces feature for client certs

### PR DESCRIPTION
**Pull request by jloehel**
## Description
* Adds `tls_client_key` option for the yatai_service
* Adds `tls_client_cert` option for the yatai_service
* Renames `client_certificate_file` to `tls_root_ca_cert` because it could cause misunderstandings.

## Motivation and Context
I'd like to deploy the YataiService with MutualTLS authentication.

## How Has This Been Tested?
I tried it against a YataiService + Nginx deployment.

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] New feature and improvements (non-breaking change which adds/improves functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Code Refactoring (internal change which is not user facing)
- [X] Documentation
- [ ] Test, CI, or build

## Component(s) if applicable
- [ ] BentoService (service definition, dependency management, API input/output adapters)
- [ ] Model Artifact (model serialization, multi-framework support)
- [ ] Model Server (mico-batching, dockerisation, logging, OpenAPI, instruments)
- [X] YataiService gRPC server (model registry, cloud deployment automation)
- [ ] YataiService web server (nodejs HTTP server and web UI)
- [ ] Internal (BentoML's own configuration, logging, utility, exception handling)
- [ ] BentoML CLI

## Checklist:

- [X] My code follows the bentoml code style, both `./dev/format.sh` and
  `./dev/lint.sh` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [X] My change requires a change to the documentation
- [X] I have updated the documentation accordingly

Signed-off-by: Jürgen Löhel <juergen.loehel@inlyse.com>

### CODE REVIEW

File: bentoml/yatai/yatai_service.py [hunk 1]
```diff
@@ -40,13 +40,20 @@ def get_yatai_service(
         scheme, addr = parse_grpc_url(channel_address)
 
         if scheme in ('grpcs', 'https'):
-            client_cacert_path = (
-                config().get('yatai_service', 'client_certificate_file')
+            tls_root_ca_cert = (
+                config().get('yatai_service', 'tls_root_ca_cert')
+                or config().get('yatai_service', 'client_certificate_file')
```
Comment (parano) [inline bentoml/yatai/yatai_service.py:45]: 👍 add an inline comment explaining this config key is meant for backward compatibility purposes?

Comment (jloehel) [reply bentoml/yatai/yatai_service.py:46]: Yes, makes sense. Sometimes that's faster than `git blame`. I added a short comment.

Comment (jloehel) [reply bentoml/yatai/yatai_service.py:46]: I haven't seen that you are using a wrapper for the ConfigParser which checks if the section, key pair is available in the configuration file. Should I add also the `client_certificate_file` to the default configuration? Do you have a process for changing the naming of configuration options?

Comment (parano) [reply bentoml/yatai/yatai_service.py:46]: @jloehel great question, we don't have a process for changing the configuration options yet, although the configparser wrapper tries to read from the config file and env var as a simple hierarchy config system, although it's quite limited. We are considering switching to use an existing framework like https://github.com/adobe/himl

File: bentoml/configuration/default_bentoml.cfg [hunk 2]
```diff
@@ -59,7 +59,9 @@ repository_base_url = {BENTOML_HOME}/repository/
 db_url = sqlite:///{BENTOML_HOME}/storage.db
 s3_endpoint_url =
 default_namespace = dev
-client_certificate_file =
+tls_root_ca_cert =
```
Comment (parano) [inline bentoml/configuration/default_bentoml.cfg:62]: Add some more notes here? e.g.
```
# Configs for securing the gRPC connection to remote YataiService
# The PEM-encoded root certificates as a byte string or local file path to the root certificate file. Leave it empty to retrieve them from a default location chosen by gRPC runtime.
tls_root_ca_cert =
# The PEM-encoded private key as a byte string or local file path to the private key file. Leave it empty if no private key should be used.
tls_client_key =
# The PEM-encoded certificate chain as a byte string or local file path to the client certificate file to use, leave it empty if no certificate chain should be used
tls_client_cert =
```

**Comment by parano** [comment]
Thank you again @jloehel, looks like the flaky test was finally passing, merging now!

**Pull request merged**